web/lib/components/flashcards/useAllSets.js

```diff
@@ -79,17 +79,20 @@ export function useAllSets({
     },
   )
 
-  const addCardToSet = useMutation(async ({ word, definition, setId }) => {
-    const { data, error } = await upsertFlashcard({
-      set_id: setId,
-      word,
-      definition,
-    })
+  const addCardToSet = useMutation(
+    async ({ word, definition, setId, userId }) => {
+      const { data, error } = await upsertFlashcard({
+        set_id: setId,
+        word,
+        definition,
+        user_id: userId,
+      })
```

Comment:
Is there any check from the database to make sure `userId` here is the current user? I'm thinking a case when someone tried to use a different ID than their own. Probably RLS and a policy are needed.

Reply:
RLS was enabled on `flashcard_sets`, so probably can also enable RLS to `flashcard`

Reply:
or `userId` of `flashcards` must be same with its `flashcard_sets`' id

Reply:
^ Of course. If user 123 created a flashcard set, then `user_id` in both `flashcard_sets` and `flashcards` have to be 123. Technically we can get user_id from `flashcards` by joining with `flashcard_sets` but it takes one more step, so that's why there is an additional `user_id` in `flashcards`.